operator_toolkit/loadshed.py

```diff
@@ -1,100 +1,49 @@
-    response = Operator.request(gains, HttpMethods.PUT,
-                                data=json.dumps(CONFIG["gains"]["loadshed"]["SoC_0_percentile"]))
-    report["set_soc_0_percentile_gains"] = Operator.status_code_msg(response)
-
-    # step 4: set SoC_25_percentile gains
-    gains = Operator.generate_url(ServiceType.RACK_CONTROL, site, path="/gains/SoC_25_percentile")
-    response = Operator.request(gains, HttpMethods.PUT,
-                                data=json.dumps(CONFIG["gains"]["loadshed"]["SoC_25_percentile"]))
-    report["set_soc_25_percentile_gains"] = Operator.status_code_msg(response)
-
-    # step 5: set SoC_50_percentile gains
-    gains = Operator.generate_url(ServiceType.RACK_CONTROL, site, path="/gains/SoC_50_percentile")
-    response = Operator.request(gains, HttpMethods.PUT,
-                                data=json.dumps(CONFIG["gains"]["loadshed"]["SoC_50_percentile"]))
-    report["set_soc_50_percentile_gains"] = Operator.status_code_msg(response)
-
-    # step 6: set suc_P gains
-    gains = Operator.generate_url(ServiceType.RACK_CONTROL, site, path="/gains/suc_P")
-    response = Operator.request(gains, HttpMethods.PUT,
-                                data=json.dumps(CONFIG["gains"]["loadshed"]["suc_P"]))
-    report["set_suc_P_gains"] = Operator.status_code_msg(response)
 
-    # step 7: delete buffer
-    buffers = Operator.generate_url(ServiceType.RACK_CONTROL, site, path="/buffers")
-    response = Operator.request(buffers, HttpMethods.DELETE)
-    report["clean_buffers"] = Operator.status_code_msg(response)
+app = typer.Typer()
 
-    # step 8: load and add the loadshed rule from path
-    rule_url = Operator.generate_url(ServiceType.VIRTUAL_OPERATOR, site, path="/schedule/rule")
-    rule = FileManager.load_file(args.rule_path)
-    rule["id_"] = f"{site}-loadshed-{datetime.today().strftime('%Y-%m-%d')}"
-    response = Operator.request(rule_url, HttpMethods.POST, json=rule)
-    report["add_loadshed_schedule_rule"] = Operator.status_code_msg(response)
 
-    return {site: report}
+def loadshed_site(site, rule_path):
+    service_manager = ServiceManager(site)
+    report = Report(site, {
+        "start_bsl_mode": service_manager.start_mode("bsl"),
+        "clear_override_target": service_manager.clear_override_target(),
+        "soc_0_percentile_gains": service_manager.set_gains(GainType.SOC_0_PERCENTILE, OperationMode.LOADSHED),
+        "soc_25_percentile_gains": service_manager.set_gains(GainType.SOC_25_PERCENTILE, OperationMode.LOADSHED),
+        "soc_50_percentile_gains": service_manager.set_gains(GainType.SOC_50_PERCENTILE, OperationMode.LOADSHED),
+        "suc_P_gains": service_manager.set_gains(GainType.SUC_P, OperationMode.LOADSHED),
+        "clear_buffers": service_manager.clear_buffers(),
+        "loadshed_schedule_rule": service_manager.add_schedule_rule(rule_path, OperationMode.LOADSHED),
+    })
```

Comment:
So initializing the `Report` by passing it a dictionary as the only argument feels strange. One really easy thing to do right now would be to just allow arbitrary keyword arguments in the initializer.
```python
class Report:
    def __init__(self, site, **report):
        self.site = site
        self.report: dict = report
```
that way here we can do
```suggestion
    report = Report(
        site,
        start_bsl_mode=service_manager.start_mode("bsl"),
        clear_override_target=service_manager.clear_override_target(),
        soc_0_percentile_gains=service_manager.set_gains(GainType.SOC_0_PERCENTILE, OperationMode.LOADSHED),
        soc_25_percentile_gains=service_manager.set_gains(GainType.SOC_25_PERCENTILE, OperationMode.LOADSHED),
        soc_50_percentile_gains=service_manager.set_gains(GainType.SOC_50_PERCENTILE, OperationMode.LOADSHED),
        suc_P_gains=service_manager.set_gains(GainType.SUC_P, OperationMode.LOADSHED),
        clear_buffers=service_manager.clear_buffers(),
        loadshed_schedule_rule=service_manager.add_schedule_rule(rule_path, OperationMode.LOADSHED),
    )
```
Just cuts out some the extraneous syntax.

Reply:
mmmm nice, I like it. Actually, i was feeling the same about this. Looks much better with this change. Thanks